src/lxc/criu.c

```diff
@@ -363,10 +363,11 @@ static void exec_criu(struct criu_opts *opts)
 	argv[argc] = NULL;
 
 	buf[0] = 0;
-	pos = buf;
+	pos = 0;
 	for (i = 0; argv[i]; i++) {
-		pos = strncat(buf, argv[i], buf + sizeof(buf) - pos);
-		pos = strncat(buf, " ", buf + sizeof(buf) - pos);
+		strncat(buf, argv[i], sizeof(buf) - pos - 1);
```

Comment:
Just out of curiosity, did you test that code on a small program or something? It looks like it would segfault when `pos += strlen(argv[i])` is larger than `sizeof(buf)`. Your code and the original code would work safely if it included a check like:
```
size_t len = 0;
for (...) {
        len += strlen(argv[i]);
        if (len + 1 > sizeof(buf))
                break; // or something similar
}
```